game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -828,12 +825,10 @@ public boolean isSatisfied(final Map<ICondition, Boolean> testedConditions, fina
         objectiveMet = false;
         changeChanceDecrementOrIncrementOnSuccessOrFailure(delegateBridge, objectiveMet, false);
       } else {
-        // there is an issue with maps using thousands of chance triggers: they are causing the cypted random source
-        // (ie: live and pbem
-        // games) to lock up or error out
+        // there is an issue with maps using thousands of chance triggers: they are causing the crypted random source
+        // (i.e.: live and pbem games) to lock up or error out
         // so we need to slow them down a bit, until we come up with a better solution (like aggregating all the chances
-        // together, then
-        // getting a ton of random numbers at once instead of one at a time)
+        // together, then getting a ton of random numbers at once instead of one at a time)
```

Comment:
This indicates a lock should be used instead of just waiting an arbitrary amount of time